angular-workspace/projects/ni/nimble-angular/src/directives/list-option/nimble-combobox-list-option.directive.ts

```diff
@@ -43,20 +43,20 @@ export class NimbleComboboxListOptionDirective implements AfterViewInit, OnDestr
     public ngAfterViewInit(): void {
         if (this.combobox) {
             this._currentTextContent = this.elementRef.nativeElement.text;
-            this.combobox.addOption(this._currentTextContent, this._modelValue);
+            this.combobox.addOption(this._currentTextContent, this._modelValue, this.elementRef);
         }
     }
 
     public ngOnDestroy(): void {
         if (this.combobox) {
-            this.combobox.removeOption(this._currentTextContent);
+            this.combobox.removeOption(this._currentTextContent, this.elementRef);
         }
     }
 
     private updateComboboxValue(value: unknown): void {
-        this.combobox!.removeOption(this._currentTextContent);
+        this.combobox!.removeOption(this._currentTextContent, this.elementRef);
         this.changeDetector.detectChanges();
         this._currentTextContent = this.elementRef.nativeElement.text;
-        this.combobox!.addOption(this._currentTextContent, value);
+        this.combobox!.addOption(this._currentTextContent, value, this.elementRef);
```

Comment:
The remove + add on value update will change the order in the optionmap, it won't align with the dom order anymore. Is that observable?
Should have an updateOptionValue or something that preserves the map order or does it not matter?

I'm now also a bit worried about reading from the nativeElement.text again here... If the text was modified it's only observed if ngValue also happens to be updated? Is there a temporal issue? It seems like the text must be assigned before ngValue for this to work since we only read the text value on ngValue update.

Reply:
I'm no longer doing add + remove, and instead just doing a direct update of the option when its model value changes. Additionally, there is a now a mutation observer that will handle if the text is changed in the DOM for keeping the internal state up-to-date for the options. These operations _do not_ affect the current display text or model value associated with the combobox.